sysversion/__init__.py

```diff
@@ -9,38 +9,66 @@
 # License, or (at your option) any later version.
 
 import re
-from subprocess import run, PIPE
-from os.path import join
+import subprocess
+from subprocess import PIPE
+from pathlib import Path
 from sys import stdin
 
 
+DEFAULT = 'etc/turnkey_version'
+
+
 def _parse_turnkey_release(version):
     m = re.match(r'turnkey-.*?-(\d.*?)-[^\d]', version)
     if m:
         return m.group(1)
 
 
+def get_debian_codename(encoding=stdin.encoding):
+    """Return Debian codename of the system (leverages lsb_release)"""
+    process = subprocess.run(['lsb_release', '-sc'], stdout=PIPE)
+    if process.returncode != 0:
+        return
+    return process.stdout.decode(encoding).strip()
+
+
 def get_turnkey_release(rootfs='/'):
     """Return release_version. On error, returns None"""
     turnkey_version = get_turnkey_version(rootfs=rootfs)
     if turnkey_version:
         return _parse_turnkey_release(turnkey_version)
 
 
-def get_turnkey_version(rootfs='/'):
+# used by turnkey-version
+def get_turnkey_version(rootfs='/', fpath=DEFAULT):
     """Return turnkey_version. On error, returns None"""
     try:
-        with open(join(rootfs, "etc/turnkey_version"), 'r') as fob:
-            return fob.read().strip()
-    except IOError:
+        p = Path(rootfs, fpath).resolve()
+        return p.read_text().strip()
+    except FileNotFoundError:
```

Comment:
I would personally say these changes to `get_turnkey_version` are superflous, creating a very minor additional resource usage for no functional difference. In my mind this would make more sense if `rootfs` was already a Path object but here it creates a new object which is used in the same way as before then dropped almost immediately.

Reply:
OK... :)